src/components/App.js

```diff
@@ -1,8 +1,196 @@
+  const [numberSelected, setNumberSelected] = useState('');
+  const [squares, setSquares] = useState([]);
+  const [wordLengths, setWordLengths] = useState([]);
+  const [remainingAlphabet, setRemainingAlphabet] = useState(alphabet);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [score, setScore] = useState(0);
+  let game;
+  let words = [];
+  let invalidWord = '';
+
+  class Game {
+    restart() {
+      setSelectNumber(false);
+      setNumberSelected('');
+      setWordLengths([]);
+      setSquares([]);
+      setWordNo(1);
+      setErrorMessage('');
+      setRemainingAlphabet(alphabet);
+      setScore(0);
+      console.clear();
+    }
+    calculateScore(words) {
+      let workScore = 20 - numberSelected;
+      for (let i = 0; i < words.length; i++) {
+        for (let j = 0; j < words[i].length; j++) {
+          workScore =
+            workScore +
+            letterPoints.find(item => {
+              return item.letter === words[i][j].toUpperCase();
+            }).point;
+      }
+    }
+    return workScore;
+    }
+  };
+  game = new Game();
+
+  const handleClick = () => {
+    // which squares contain 1's,2's,3's,.....
+    let wordN;
+    let result = '';
+    for (let i = 1; i < wordNo + 1; i++) {
+      wordN = [];
+      for (let j = 0; j < squares.length; j++) {
+        wordN = squares.filter(event => {
+          if (event.wordNums.length === 1) {
+            if (event.wordNums[0] === i) {
+              return true;
+            } else {
+              return false;
+            }
+          } else if (event.wordNums.length === 2) {
+            if (event.wordNums[0] === i || event.wordNums[1] === i) {
+              return true;
+            } else {
+              return false;
+            }
+          } else {
+            return false;
+          }
+        });
+      }
+      // combine words from board
+      result = '';
+      for (let k = 0; k < wordN.length; k++) {
+        result = result + wordN[k].letter;
+      }
+      words.push(result);
+    }
+    verifyBoard(words);
+  };
+
+  function checkWords() {
+    let allValid = true;
+    //let invalidWord = '';
+    for (let j = 0; j < words.length; j++) {
+      if (validWord(words[j]) === false) {
+        allValid = false;
+        invalidWord = words[j];
+        break;
+      }
+    }
+    return allValid
+  }
+
+  function verifyBoard(words) {
+    let workErrorMessage = '';
+    // Are all squares filled in?
+    for (let j = 0; j < squares.length; j++) {
+      if (squares[j].letter === '') {
+        workErrorMessage = 'Fill in all squares';
+        setErrorMessage(workErrorMessage);
+        return;
+      }
+    }
+    // are all words real words
+    if (!checkWords()) {
+      workErrorMessage = invalidWord + ' is not valid';
+    } else {
+      workErrorMessage = 'You win!!!';
+      setScore(game.calculateScore(words));
+    }
+    setErrorMessage(workErrorMessage);
+  }
+
+  function validWord(word) {
```

Comment:
Helpers like this should be moved to another file like a `utils.js` so we can separate the basic logic functions from the UI code

Reply:
Done.